tests/test_predicates.py

```diff
@@ -43,7 +46,7 @@ def test_binary_predicate_exceptions(self):
         p2 = [(339, 207), (280, 311), (460, 138), (399, 242), (459, 277),
               (459, 415), (399, 381), (519, 311), (520, 242), (519, 173),
               (399, 450), (339, 207)]
-        self.assertRaises(TopologicalError, Polygon(p1).within, Polygon(p2))
+        self.assertRaises(pygeos.GEOSException, Polygon(p1).within, Polygon(p2))
```

Comment:
Currently, shapely has a bunch of custom error classes in https://github.com/Toblerity/Shapely/blob/master/shapely/errors.py. But in PYGEOS, for now we just have a single GEOSException for errors that are directly raised from one of the GEOS functions.

Reply:
Raising pygeos exceptions is just an intermediate state of the API then?

Reply:
Yes, the fact that it is a "pygeos" exception is intermediate state. But basically everywhere you see `pygeos` you need to mentally replace it with `shapely` (as in one of the next steps we will move pygeos into shapely and replace the pygeos imports with internal shapely imports). 

So the "real" question here is if we are, long term, fine with having only a generic `shapely.GEOSException`, or if we want to have more fine grained error *classes* (TopologicalError, WKBReadingError, WKTReadingError). 
Note that this is only about the class of the exception, also the generic GEOSException error class will still have the custom, informative error messages.

Reply:
Opened https://github.com/Toblerity/Shapely/issues/991 to have a dedicated place to discuss this topic.